Use the Az module instead of AzureRM

paddy74 opened the pull request:
AzureRM is deprecated and not compatible with PowerShell 7. As this script is referenced in official documentation, it should be updated.
This commit also updates variable names where they are retrieved such as `.SubscriptionName` to `.DisplayName` where the former no longer exists.

paddy74 committed:
Use the Az module instead of AzureRM

AzureRM is deprecated and not compatible with PowerShell 7. As this script is referenced in official documentation, it should be updated.
This commit also updates variable names where they are retrieved such as `.SubscriptionName` to `.DisplayName` where the former no longer exists.

dcaro commented on spncreation.ps1:
```diff
@@ -41,57 +41,57 @@ function Get-Password
-$connectionName = $azureSubscription.SubscriptionName
+Import-Module -Name Az.Accounts
+Write-Output "Provide your credentials to access your Azure subscription $subscriptionName" -Verbose
+Login-AzAccount -SubscriptionName $subscriptionName -EnvironmentName $environmentName
```
```suggestion
Connect-AzAccount -SubscriptionName $subscriptionName -EnvironmentName $environmentName
```

dcaro commented on spncreation.ps1:
```diff
@@ -41,57 +41,57 @@ function Get-Password
 Write-Output "Creating a new Application in AAD (App URI - $identifierUri)" -Verbose
 $password = Get-Password
-$azureAdApplication = New-AzureRmADApplication -DisplayName $displayName -HomePage $homePage -IdentifierUris $identifierUri -Password $password -Verbose
+$azureAdApplication = New-AzADApplication -DisplayName $displayName -HomePage $homePage -IdentifierUris $identifierUri -Password $password -Verbose
```
This command will not work with Az PowerShell 7 and above. It is using Microsoft Graph sot the password cannot be set and the identifier Uri must use a verified domain in of the organization or its subdomain.

```suggestion
$azureAdApplication = New-AzADApplication -DisplayName $displayName -HomePage $homePage -Verbose
```

dcaro commented on spncreation.ps1:
```diff
@@ -41,57 +41,57 @@ function Get-Password
 $password = Get-Password
-$azureAdApplication = New-AzureRmADApplication -DisplayName $displayName -HomePage $homePage -IdentifierUris $identifierUri -Password $password -Verbose
+$azureAdApplication = New-AzADApplication -DisplayName $displayName -HomePage $homePage -IdentifierUris $identifierUri -Password $password -Verbose
 $appId = $azureAdApplication.ApplicationId
```
```suggestion
$appId = $azureAdApplication.AppId
```

dcaro commented on spncreation.ps1:
```diff
@@ -41,57 +41,57 @@ function Get-Password
 Start-Sleep 20
 Write-Output "Assigning role ($spnRole) to SPN App ($appId)" -Verbose
-New-AzureRmRoleAssignment -RoleDefinitionName $spnRole -ServicePrincipalName $appId
+New-AzRoleAssignment -RoleDefinitionName $spnRole -ServicePrincipalName $appId
```
```suggestion
New-AzRoleAssignment -RoleDefinitionName $spnRole -ServicePrincipalName $spn.AppId
```

dcaro commented on spncreation.ps1:
```diff
@@ -15,20 +15,20 @@ param
 {
     $currentAzurePSVersion = Get-AzureCmdletsVersion
-    $azureVersion511 = New-Object System.Version(5, 1, 1)
+    $minAzurePSVersion = New-Object System.Version(5, 6, 0)
```
Az PowerShell version 8 is now available, I would recommend considering the most recent version of this module for this script.
```suggestion
    $minAzurePSVersion = New-Object System.Version(8, 0, 0)
```

dcaro reviewed (commented):
I made some comments to make this PR compatible with Az PowerShell version 8.

stevevi commented:
Thank you @paddy74 and @dcaro for your input.  I updated the article and moved the PS script into the main body of the article for easier maintenance.

https://docs.microsoft.com/en-us/azure/azure-government/connect-with-azure-pipelines#generate-a-service-principal

This standalone dev repo is no longer referenced from the updated article.  Feel free to close the PR.  Also, my apologies @paddy74 that it took such a long time to act on your initial PR.  The software dev who originally wrote the script changed roles and this repo was no longer monitored actively.

paddy74 committed:
Correct variable names via suggested changes

Apply suggestions from code review.

Co-authored-by: Damien Caro <dcaro@microsoft.com>

paddy74 committed:
Set min pwsh version to 8 via suggested change

Apply suggestions from code review

Co-authored-by: Damien Caro <dcaro@microsoft.com>

paddy74 committed:
Remove unsupported New-AzAdApplication parameters

Apply suggestions from code review.
The command is using Microsoft Graph so the password cannot be set and the identifier Uri must use a verified domain in the organization or its subdomain.

Co-authored-by: Damien Caro <dcaro@microsoft.com>

paddy74 commented:
@dcaro I've applied the recommended changes.
Although this repository is no longer monitored or referenced by the documentation, individuals may still end up referencing it. As such, I recommend these changes be merged.
A separate PR should be opened to indicate in the `README.md` that this repository is deprecated and to direct users to the documentation. If possible, I also recommend the "deprecated" tag be added to this repository.

stevevi commented:
Thanks @paddy74 for your input.  I'll defer to @yujhongmicrosoft who is the repo owner for further action.
